pkg/cloudprovider/servicecontroller/servicecontroller.go

```diff
@@ -225,44 +225,25 @@ func (s *ServiceController) createLoadBalancerIfNeeded(namespacedName types.Name
 		glog.Infof("LB already exists and doesn't need update for service %s", namespacedName)
 		return nil, notRetryable
 	}
-	if cachedService != nil {
```

Comment:
To be more specific - it'll break us in the case where some aspect of the service definition has changed (such as the port numbers) that requires modifying the load balancer, which we currently implement by tearing it down and recreating it.